commands/util/pvp-rank.js

```diff
@@ -0,0 +1,161 @@
+const log = require('loglevel').getLogger('Counters'),
+  Commando = require('discord.js-commando'),
+  {MessageEmbed} = require('discord.js'),
+  {CommandGroup} = require('../../app/constants'),
+  fetch = require('node-fetch'),
+  cheerio = require('cheerio'),
+  Utility = require('../../app/utility');
+
+class PvPRankingData {
+  constructor(pokemon, league, atk, def, sta, filter){
+    this.inputPokemon = pokemon;
+    this.inputLeague = league;
+    this.inputATKiv = atk;
+    this.inputDEFiv = def;
+    this.inputSTAiv = sta;
+    if(typeof(filter) == 'undefined'){
+      this.inputFilter = "0";
+    }
+    else{
+      this.inputFilter = filter;
+    }
+  }
+
+  async scrape(){
+    String.prototype.titleCase = function () {
+      return this.replace(/\w\S*/g, function(str) {return str.charAt(0).toUpperCase() + str.substr(1).toLowerCase();});
+    };
+    this.cpLeague;
+    let ivFilter;
+    if(this.inputLeague == 'ultra'){
+      this.cpLeague = "2500";
+    }
+    else {
+      this.cpLeague = "1500";
+    }
+    if(this.inputFilter != "10"){

```

Comment:
What happens if `this.inputFilter` is `10`? You don't have a default value for `ivFilter` if this set to 10 (meaning the url becomes `&min_iv=undefined`.

Reply:
This code will be used in future versions when I include filters as args to the command to provide an optional minimum IV. Right now, you're right- it just looks like unfinished code. But right now it's impossible AFAIK for inputFilter to be equal to 10. All called objects of the class leave ivFilter's parameter empty in the constructor.

Reply:
Sounds good.